cmd/hub/mcscontroller/main.go

```diff
@@ -73,15 +78,15 @@ func main() {
 	flag.Parse()
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
-	if namespace == "" {
+	if namespace == corev1.NamespaceAll {
 		ns, err := getWatchedNamespace()
 		if err != nil {
 			setupLog.Error(err, "Failed to get watched namespace from environment")
 		}
 		namespace = ns
 	}
 
-	if namespace == "" {
+	if namespace == corev1.NamespaceAll {
```

Comment:
does not make sense to fail here after the previous block with getWatchedNamespace(). you should put this block before the other one..